src/parser/jobs/ast/modules/Lightspeed.tsx

```diff
@@ -0,0 +1,231 @@
+import {filter} from 'parser/core/filter'
+import {dependency} from 'parser/core/Injectable'
+import {Data} from 'parser/core/modules/Data'
+import {Timeline} from 'parser/core/modules/Timeline'
+import React, {Fragment} from 'react'
+import {Icon} from 'semantic-ui-react'
+import DISPLAY_ORDER from './DISPLAY_ORDER'
+
+const BASE_GCDS_PER_WINDOW = 6
+
+class LIGHTSPEED_Window {
+	start: number
+	end?: number
+
+	rotation: Array<Events['action']> = []
+	gcdCount: number = 0
+	mpSavings: number = 0
+	trailingGcdEvent?: Events['action']
+
+	buffsRemoved: Array<Status['id']> = []
+	deathTruncated: boolean = false
+
+	constructor(start: number) {
+		this.start = start
+	}
+}
+
+// in this module we only want to track Lightspeed windows opened by
+// the character selected for analysis. windows that clip into
+// AST Lightspeed will be marked.
+// Used DNC Technicalities as basis for this module. Rewritten from previous module for consistency purposes
+export default class Lightspeed extends Analyser {
+	static override handle = 'Lightspeed'
+	static override title = t('ast.Lightspeed.title')`Lightspeed`
+	static override displayOrder = DISPLAY_ORDER.LIGHTSPEED
+
+	@dependency private timeline!: Timeline
+	@dependency private data!: Data
+
+	private history: LIGHTSPEED_Window[] = []
+
+	override initialise() {
+		const lightspeedFilter = filter<Event>().status(this.data.statuses.LIGHTSPEED.id)
+
+		this.addEventHook(lightspeedFilter.type('statusApply')
+			.target(this.parser.actor.id), this.tryOpenWindow)
+		this.addEventHook(lightspeedFilter.type('statusRemove')
+			.target(this.parser.actor.id), this.tryCloseWindow)
+
+		this.addEventHook(filter<Event>().source(this.parser.actor.id).type('action'), this.onCast)
+	}
+
+	private tryOpenWindow(event: Events['statusApply']): LIGHTSPEED_Window {
+		const lastWindow: LIGHTSPEED_Window | undefined = _.last(this.history)
+
+		// If that happens, re-open the last window and keep tracking
+		if (lastWindow != null) {
+			if (!lastWindow.end) {
+				return lastWindow
+			}
+			if (lastWindow.end === event.timestamp) {
+				lastWindow.end = undefined
+				return lastWindow
+			}
+		}
+
+		const newWindow = new LIGHTSPEED_Window(event.timestamp)
+		this.history.push(newWindow)
+		return newWindow
+	}
+
+	private tryCloseWindow(event: Events['statusRemove']) {
+		const lastWindow: LIGHTSPEED_Window | undefined = _.last(this.history)
+
+		if (lastWindow == null) {
+			return
+		}
+
+		// Cache whether we've seen a buff removal event for this status, just in case they happen at exactly the same timestamp
+		lastWindow.buffsRemoved.push(event.status)
+
+		if (this.isWindowOkToClose(lastWindow)) {
+			lastWindow.end = event.timestamp
+		}
+	}
+
+	// Make sure all applicable statuses have fallen off before the window closes
+	private isWindowOkToClose(window: LIGHTSPEED_Window): boolean {
+		if (!window.buffsRemoved.includes(this.data.statuses.LIGHTSPEED.id)) {
+			return false
+		}
+		return true
```

Comment:
Easier to read, nbd.

```suggestion
		return !window.buffsRemoved.includes(this.data.statuses.LIGHTSPEED.id)) ? false : true
```

Reply:
I just removed this function and just replaced the section that was used with the literal check since it just checks false/true anyway